src/profile/helpers/__tests__/editData.test.ts

```diff
@@ -102,5 +165,187 @@ describe('editData.ts ', () => {
+    it(`returns new list when new profile data has no items.'`, () => {
+      const emaillessProfileData: ProfileRoot = getMyProfile();
+      (emaillessProfileData.myProfile as Mutable<ProfileData>).emails = null;
+      const { updateData, getEditData } = createEditorForDataType(
+        myProfile,
+        'emails'
+      );
+      updateData(emaillessProfileData);
+      expect(getEditData()).toHaveLength(0);
+    });
+  });
+  describe(`updateItems `, () => {
+    const dataType = 'phones';
+    let editorFunctions: EditFunctions;
+    let editDataList: EditData[];
+    let sources: EditDataProfileSource[];
+    beforeEach(() => {
+      editorFunctions = createEditorForDataType(myProfile, dataType);
+      editDataList = editorFunctions.getEditData();
+      sources = pickSources(myProfile.myProfile as ProfileData, dataType);
+    });
+    it(`returns null when edit data has no items with saving set to 'value'`, () => {
+      const nullList = updateItems(editDataList, sources, dataType);
+      expect(nullList).toBeNull();
+    });
+    it(`returns null until item.value and source match and item.saving='value'`, () => {
+      const item = editDataList[0] as Mutable<EditData>;
+      const value = item.value as Mutable<PhoneValue>;
+      const source = sources[0] as Mutable<PhoneNode>;
+      item.saving = 'value';
+      value.phone = createNewValue('phones');
+      const nullList = updateItems(editDataList, sources, dataType);
+      expect(nullList).toBeNull();
+      source.phone = value.phone;
+      const newList = updateItems(
+        editDataList,
+        sources,
+        dataType
+      ) as EditData[];
+      expect(newList).toHaveLength(editDataList.length);
+      const updatedItem = newList[0];
+      // old item has not changed
+      expect(item.saving).toEqual('value');
+      // change saving so item will match updatedItem
+      item.saving = undefined;
+      expect(updatedItem).toEqual(item);
+      expect(updatedItem.saving).toEqual(undefined);
+    });
+    it(`returns null until item.primary and source.primary match and item.saving='set-primary'`, () => {
+      const item = editDataList[0] as Mutable<EditData>;
+      const source = sources[0] as Mutable<PhoneNode>;
+      item.saving = saveTypeSetPrimary;
+      item.primary = true;
+      source.primary = false;
+      const nullList = updateItems(editDataList, sources, dataType);
+      expect(nullList).toBeNull();
+      source.primary = true;
+      const newList = updateItems(
+        editDataList,
+        sources,
+        dataType
+      ) as EditData[];
+      expect(newList).toHaveLength(editDataList.length);
+      const updatedItem = newList[0];
+      // change saving so item will match updatedItem
+      expect(updatedItem).toEqual({ ...item, saving: undefined });
+      expect(updatedItem.saving).toEqual(undefined);
+    });
+    it(`Added new items are not lost and are updated from source'`, () => {
```

Comment:
Should we skip the this test for now or add todo or jira id? It might be confusing looking test results saying this test pass when it does not do anything yet?

Reply:
Ok, I left it there as a placeholder. Adding new items is in next PR. I'll delete this.

Reply:
Removed